promql/query_logger.go

```diff
@@ -83,6 +86,23 @@ func logUnfinishedQueries(filename string, filesize int, logger log.Logger) {
 	}
 }
 
+type mmapedFile struct {
+	f io.Closer
+	m mmap.MMap
+}
+
+func (f *mmapedFile) Close() error {
+	err := f.m.Unmap()
+	if fErr := f.f.Close(); fErr != nil && err == nil {
```

Comment:
we could use two err variables and `errors.Join` them (nil is ignored)

Reply:
Nice suggestion, thanks!